langchain4j-oracle-database/src/test/java/dev/langchain4j/store/embedding/oracle/TestDataSource.java

```diff
@@ -0,0 +1,118 @@
+package dev.langchain4j.store.embedding.oracle;
+
+import oracle.jdbc.OracleConnection;
+import oracle.jdbc.OracleConnectionWrapper;
+
+import javax.sql.DataSource;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLTimeoutException;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+
+/**
+ * A DataSource implementation used by tests in this module. A simplistic connection pool is implemented so that
+ * {@link #getConnection()} will return the same connection on successive calls. This allows tests to complete much
+ * faster than they would if a new connection was created on each call. More sophisticated connection pools are readily
+ * available, but using this pool will avoid introducing a new dependency. Not introducing dependencies is one of the
+ * guidelines listed in the CONTRIBUTING.md document.
```

Comment:
> Not introducing dependencies is one of the guidelines listed in the CONTRIBUTING.md document.

This requirement does not apply to test code, please feel free to use extra dependencies for test

Reply:
Latest commit has tests using Universal Connection Pool (UCP), and my little TestDataSource is removed. My team develops UCP in tandem with the Oracle JDBC Driver, so it should be a good option; We are constantly running integration tests to ensure that UCP is compatible with the latest builds of our driver.